spec/services/create_newsletter_signup_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+RSpec.describe CreateNewsletterSignup do
+
+  describe "newsletter signup object" do
+    it "accepts a first name, last name, and email" do
+      signup = CreateNewsletterSignup.new(
+          :newsletter_signup => { :first_name => "Will", :last_name => "Blake", :email => "test@test.com" }
+        )
+
+      expect(signup.newsletter_signup[:first_name]).to eq('Will')
+      expect(signup.newsletter_signup[:last_name]).to eq('Blake')
+      expect(signup.newsletter_signup[:email]).to eq('test@test.com')
+    end
+  end
+
+  describe "#perform" do
+    it "sends a signup request to SendGrid api" do
+      service = CreateNewsletterSignup.new(
+        :newsletter_signup => { :first_name => "Will", :last_name => "Blake", :email => "test@test.com" }
+      ).perform
+
+      response = JSON.parse(service.body)
+      expect(response.keys[0]).to eq('job_id')
```

Comment:
I would like to suggest a different approach to this test. Checkout Webmock! We should never allow API requests to hit a real live endpoint, even with test keys, from the test suite.

Reply:
I switched out the test to use webmock instead of hitting the live sendgrid api and read about the reasons it's not a good idea to do that. I'm sure the test could use more coverage, especially now that I wrote some error handling for the keyword arguments. Let me know!